integration/query_array_test.go

```diff
@@ -137,3 +140,96 @@ func TestQueryArraySize(t *testing.T) {
 		})
 	}
 }
+
+func TestQueryArrayDotNotation(t *testing.T) {
+	t.Parallel()
+	providers := []shareddata.Provider{shareddata.Composites}
+	ctx, collection := setup(t, providers...)
+
+	_, err := collection.InsertMany(ctx, []any{
+		bson.D{{"_id", "array-double"}, {"value", bson.A{float64(1)}}},
+		bson.D{{"_id", "array-strings"}, {"value", bson.A{"foo", "bar", "baz"}}},
+		bson.D{
+			{"_id", "document-array-field"},
+			{"value", bson.D{{"array", bson.A{int32(0), nil}}}},
+		},
+		bson.D{
+			{"_id", "document-document-field"},
+			{"value", bson.D{{"document", bson.D{{"foo", nil}}}}},
+		},
+	})
+	require.NoError(t, err)
+
+	for name, tc := range map[string]struct {
+		filter      bson.D
+		expectedIDs []any
+		err         *mongo.CommandError
+	}{
+		"PositionDouble": {
+			filter:      bson.D{{"value.0", bson.D{{"$type", "double"}}}},
+			expectedIDs: []any{"array-double"},
+		},
+		"PositionString": {
+			filter:      bson.D{{"value.0", bson.D{{"$type", "string"}}}},
+			expectedIDs: []any{"array-strings"},
+		},
+		"PositionNull": {
+			filter:      bson.D{{"value.0", bson.D{{"$type", "null"}}}},
+			expectedIDs: []any{"array-null", "array-three-reverse"},
+		},
+		"PositionRegex": {
+			filter:      bson.D{{"value.0", primitive.Regex{Pattern: "foo"}}},
+			expectedIDs: []any{"array-strings"},
+		},
+		"NoSuchFieldPosition": {
+			filter:      bson.D{{"value.some.0", bson.A{42}}},
+			expectedIDs: []any{},
+		},
+		"Field": {
+			filter:      bson.D{{"value.array", int32(42)}},
+			expectedIDs: []any{"document-composite", "document-composite-reverse"},
+		},
+		"FieldPosition": {
+			filter:      bson.D{{"value.array.0", int32(42)}},
+			expectedIDs: []any{"document-composite", "document-composite-reverse"},
+		},
+		"FieldPositionQuery": {
+			filter:      bson.D{{"value.array.0", bson.D{{"$lt", int32(42)}}}},
+			expectedIDs: []any{"document-array-field"},
+		},
+		"FieldPositionQueryNonArray": {
+			filter:      bson.D{{"value.document.0", bson.D{{"$lt", int32(42)}}}},
+			expectedIDs: []any{},
+		},
+		"FieldPositionField": {
+			filter:      bson.D{{"value.array.2.foo", "bar"}},
+			expectedIDs: []any{},
+		},
+
+		"FieldPositionQueryRegex": {
+			filter: bson.D{{"value.array.0", bson.D{{"$lt", primitive.Regex{Pattern: "^$"}}}}},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "Can't have RegEx as arg to predicate over field 'value.array.0'.",
```

Comment:
It's already been covered, no?

Reply:
It's not been covered with dot notation field.

Reply:
I don't get the problem here. I'm not covering regex error, I'm covering dot notation. It turned out that you can't use the query as we have in that test.